python/ray/data/datasource/file_meta_provider.py

```diff
@@ -196,11 +196,6 @@ def expand_paths(
         paths: List[str],
         filesystem: "pyarrow.fs.FileSystem",
     ) -> Tuple[List[str], List[Optional[int]]]:
-        logger.warning(
-            f"Skipping expansion of {len(paths)} path(s). If your paths contain "
-            f"directories or if file size collection is required, try rerunning this "
-            f"read with `meta_provider=DefaultFileMetadataProvider()`."
-        )
```

Comment:
I've removed this warning because I think it'll lead to a better UX for the user.

If the user calls `ray.data.read_datasource(ImageFolderDatasource(), ...)`, a warning is raised:

> (_prepare_read pid=43380) 2022-05-11 02:11:33,272       WARNING [file_meta_provider.py:199](http://file_meta_provider.py:199/) -- Skipping expansion of 2 path(s). If your paths contain directories or if file size collection is required, try rerunning this read with `meta_provider=DefaultFileMetadataProvider()`.

This may confuse the user because:
1. It has nothing to do with the user's problem
2. It leaks an implementation detail
3. It can't be resolved

Because `ImageFolderDatasource` is user-facing and `FastFileMetadataProvider` is a developer API, I think it's overall better to remove the warning.

Reply:
Hey please split this change into a separate PR and raise it in the data team for discussion.

Reply:
Done. See #24909